main.go

```diff
@@ -44,11 +44,7 @@ func main() {
 		context.Rest.SetRoutes(c.Routes())
 	}
 
-	// Store self information in the store if GlusterD is coming up for
-	// first time
-	if context.Restart == false {
-		peer.AddSelfDetails()
-	}
+	peer.AddSelfDetails()
```

Comment:
Yes. I can see the node details in etcd store soon after glusterd start. One small caveat is that on every restart of glusterd, the etcd entry attributes createdIndex and modifiedIndex numeric value is bumped up.